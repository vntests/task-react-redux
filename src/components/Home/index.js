import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';

import QuoteList from '../../containers/QuoteList';

import { Container, Row, Col } from 'reactstrap';

import './index.css';

const Home = ({ t }) => (
  <div>
    <h1>{t('home.title')}</h1>

    <Container>
      <Row>
        <Col>
          <QuoteList />
        </Col>
      </Row>
    </Container>
  </div>
);

Home.propTypes = {
  t: PropTypes.func.isRequired
};

export default translate('translations')(Home);
