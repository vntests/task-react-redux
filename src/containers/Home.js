import React, { Component } from 'react';
import { connect } from 'react-redux';
import { translate } from 'react-i18next';

import Home from '../components/Home';

class HomeContainer extends Component {
  render() {
    const { t } = this.props;

    return (
      <Home
        title={t('home.title')}
      />
    );
  }
}

export default connect()(translate('translations')(HomeContainer));
