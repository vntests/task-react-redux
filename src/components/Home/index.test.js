import React from 'react';
import renderer from 'react-test-renderer';

import Home from './';

jest.mock('react-i18next', () => ({
  translate: () => Component => {
    Component.defaultProps = { ...Component.defaultProps, t: () => '' };
    return Component;
  }
}));

jest.mock('../../containers/QuoteList', () => 'QuoteList');

test('<Home /> should render', () => {
  const component = renderer.create(
    <Home />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
