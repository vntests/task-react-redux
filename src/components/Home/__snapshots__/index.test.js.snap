// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render 1`] = `
<div>
  <h1>
    
  </h1>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <QuoteList />
      </div>
    </div>
  </div>
</div>
`;
